src/panels/config/energy/components/ha-energy-grid-settings.ts

```diff
@@ -449,11 +449,7 @@ export class EnergyGridSettings extends LitElement {
       ),
     };
 
-    try {
-      await this._savePreferences(preferences);
-    } catch (err: any) {
-      showAlertDialog(this, { title: `Failed to save config: ${err.message}` });
-    }
+    await this._deleteSource(preferences);
```

Comment:
```suggestion
    const cleanedPreferences = this._removeEmptySources(preferences)
    await this._savePreferences(cleanedPreferences);
```

Can we have something like this? It's confusing to call a method named `_deleteSource` with preferences as a parameter.

Reply:
Sure, done.